File: groups/group1/DivAlgo.java
import java.util.Scanner;

public class DivAlgo {
   static void print(Object object) {
        System.out.print(object);
    }

    static void println(Object object) {
        System.out.println(object);
    }

    static void calculateDivAlgo(int m, int n) {
        int q = m / n;
        int r = m % n;

        println("\nSOLUTION:");
        println(m + " = " + n + " (" + q + ") + " + r);
        println("");
        println("The dividend is " + m);
        println("The divisor is " + n);
        println("The quotient is " + q + " and the remainder is " + r);
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        // FIRST INPUT
        print("Enter the first integer: ");
        if (!sc.hasNextInt()) {                          // hindi integer
            println("Invalid input. Please enter an integer.");
            sc.close();
            return;
        }
        int firstInput = sc.nextInt();                   // ngayon lang tayo bumabasa
        if (firstInput <= 0) {                           // ≤ 0? error agad
            println("Invalid input. Please enter a positive integer.");
            sc.close();
            return;
        }

        // SECOND INPUT
        print("Enter the second integer: ");
        if (!sc.hasNextInt()) {                          // hindi integer
            println("Invalid input. Please enter an integer.");
            sc.close();
            return;
        }
        int secondInput = sc.nextInt();
        if (secondInput <= 0) {                          // ≤ 0? error agad
            println("Invalid input. Please enter a positive integer.");
            sc.close();
            return;
        }

        int m = Math.max(firstInput, secondInput);
        int n = Math.min(firstInput, secondInput);

        calculateDivAlgo(m, n);
        sc.close();
    }
}

File: groups/group1/DivAlgo.js
function calculateDivAlgo(m, n) {
  const q = Math.floor(m / n);
  const r = m % n;

  return [
    "",
    "SOLUTION:",
    `${m} = ${n} (${q}) + ${r}`,
    "",
    `The dividend is ${m}`,
    `The divisor is ${n}`,
    `The quotient is ${q} and the remainder is ${r}`,
  ].join("\n");
}

function parsePositiveInt(rawValue) {
  const trimmed = String(rawValue ?? "").trim();

  if (!/^-?\d+$/.test(trimmed)) {
    throw new Error("Invalid input. Please enter an integer.");
  }

  const value = Number.parseInt(trimmed, 10);

  if (value <= 0) {
    throw new Error("Invalid input. Please enter a positive integer.");
  }

  return value;
}

function main(inputs) {
  const firstInput = parsePositiveInt(inputs[0]);
  const secondInput = parsePositiveInt(inputs[1]);

  const m = Math.max(firstInput, secondInput);
  const n = Math.min(firstInput, secondInput);

  return calculateDivAlgo(m, n);
}
